### PR TITLE
Jakarta Interceptors 2.0

### PR DESCRIPTION
**Pull request by arjantijms**
Signed-off-by: arjantijms <arjan.tijms@gmail.com>

## Specification PR template
When creating a specification project release review, create PRs with the content defined as follows.

Include the following in the PR:
- [X] A directory in the form wombat/x.y where x.y is the release major.minor version, and the directory contains the following.
- [X] Specification PDF in the form of wombat-spec-x.y.pdf
- [X] Specification HTML in the form of wombat-spec-x.y.html
- [X] A specification page named _index.md following the template at:
      https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md
- [ ] For a Progress Review, that sufficient progress has been made on a Compatible Implementation and TCK, to ensure that the spec is implementable and testable.
- [X] For a Release Review, a summary that a Compatible Implementation is complete, passes the TCK, and that the TCK includes sufficient coverage of the specification.
The TCK users guide MUST include the instructions to run the compatible implementations used to validate the release.
Instructions MAY be by reference.
- [X] The URL of the OSSRH staging repository for the api, javadoc:
      https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/interceptor/jakarta.interceptor-api/2.0.0/
- [X] The URL of the staging directory on downloads.eclipse.org for the proposed EFTL TCK binary:
      https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-jakartaeetck-9.0.0.zip
- [X] The URL of the compatibility certification request issue:
      https://github.com/eclipse-ee4j/interceptor-api/issues/71
- [X] Specification JavaDoc in the wombat/x.y/apidocs directory. 
If desired, an optional second PR can be created to contain just the JavaDoc in the `apidocs` directory.

### CODE REVIEW

**Comment by arjantijms** [comment]
Note that Interceptors doesn't have its own TCK, but piggybacks on the platform TCK.

**Comment by kwsutter** [comment]
Assigning @kazumura as the Mentor for this Spec PR.

**Comment by kazumura** [comment]
# Spec Review Checklist

1. Spec PR
  - [x] PR uses [template](https://github.com/jakartaee/specifications/blob/master/pull_request_template.md)
  - [x] Directory of form {spec}/x.y
  - [x] PDF of form {spec}-spec-x.y.pdf ("-spec" preferred but not required)
  - [x] HTML of form {spec}-spec-x.y.html ("-spec" preferred but not required)
  - [x] Index page {spec}/x.y/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_page_template.md)
  - [x] Index page {spec}/_index.md following [template](https://github.com/jakartaee/specification-committee/blob/master/spec_index_template.md)
  - [x] No other files (e.g., no jakarta_ee_logo_schooner_color_stacked_default.png)
  - [ ] Staging repository link of the form https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/{spec}/jakarta-{spec}-api/x.y.z/
  - [ ] EFTL TCK link of the form http://download.eclipse.org/.../+.zip
  - [ ] Compatibility certification link of the form https://github.com/eclipse-ee4j/{project}/#{issue}
  - [ ] (Optional) Second PR for just apidocs

2. _index.md
  - [ ] Link to spec pdf
  - [ ] Link to spec html
  - [ ] Link to apidocs
  - [ ] Link to final TCK download zip file of the form https://download.eclipse.org/jakartaee/{spec}/x.y/*{spec}-tck-x.y.z.zip
  - [ ] Link to API jar file of the form https://search.maven.org/artifact/jakarta.{spec}/jakarta.{spec}-api/x.y.z/jar
  - [ ] Name of and link to at least one Compatible Implementation

3. javadocs
  - [ ] Footer contains Eclipse copyright and link to license
  - [ ] ESFL license is included, usually as doc-files/speclicense.html
  - [ ] no META-INF directory in PR
  - [ ] javadocs-jar artifact matches apidocs (optional for this release)

4. Spec PDF
  - [ ] Correct spec title
  - [ ] Version number of the form x.y, not x.y.z
  - [ ] Correct Eclipse copyright line
  - [ ] No DRAFT or SNAPSHOT
  - [ ] Correct Logo

5. Spec HTML
  - [ ] Same as PDF

6. TCK zip file
  - [ ] README file (optional for this release)
  - [ ] EFTL license file, preferably named LICENSE.md
  - [ ] User's Guide (or equivalent documentation)
  - [ ] How to test the Compatible Implementation(s) listed in _index.md above with the TCK (may be in UG)

7. TCK User's Guide (or equivalent documentation)
  - [ ] Software requirements listed
  - [ ] Installation and configuration described
  - [ ] How to run tests
  - [ ] Where to file challenges

8. Compatibility certification request
  - [ ] Request follows [template](https://github.com/jakartaee/specification-committee/blob/master/compatibility-certification-request.md)
  - [ ] SHA-256 fingerprint matches staged TCK zip file
  - [ ] Request issue has `certification` label.

9. TCK results summary
  - [ ] Page is hosted by Compatible Implementation project
  - [ ] Includes all information from certification request
  - [ ] Summary includes number of tests passed, failed, errors
  - [ ] SHA-256 fingerprint matches staged TCK zip file on cert request
 
10. Update Jakarta EE API jar
  - [ ] Update the Jakarta EE API jar by submitting a PR to the jakartaee-api project that updates the version number of your API jar file.

**Comment by kwsutter** [comment]
Tagging with "Platform" to indicate it's dependency on the Platform TCK.

**Comment by arjantijms** [comment]
Thanks @kazumura for the extra pair of eyes!

File: interceptors/2.0/_index.md [hunk 1]
```diff
@@ -9,17 +9,17 @@ and other managed classes.
 
 * [Jakarta Interceptors 2.0 Release Record](https://projects.eclipse.org/projects/ee4j.interceptors/releases/2.0.0)
   * [Jakarta EE Platform 9 Release Plan](https://eclipse-ee4j.github.io/jakartaee-platform/jakartaee9/JakartaEE9ReleasePlan)
-* [Jakarta Interceptors 2.0 Specification Document](./interceptors-spec-2.0-SNAPSHOT.pdf) (PDF)
-* [Jakarta Interceptors 2.0 Specification Document](./interceptors-spec-2.0-SNAPSHOT.html) (HTML)
+* [Jakarta Interceptors 2.0 Specification Document](./interceptors-spec-2.0.pdf) (PDF)
```
Comment (jeanouii) [inline interceptors/2.0/_index.md:12]: shouldn't the file be named jakarta-....?

**Comment by jeanouii** [comment]
There is one mention to `javax.xml.rpc.handler.MessageContext` in the api for InvocationContext.

It should be jakarta.xml.rpc.handler.MessageContext

File: interceptors/2.0/_index.md (hunk 1 above)
Comment (arjantijms) [reply interceptors/2.0/_index.md:12]: I think not, the other specs don't have that either:

* https://jakarta.ee/specifications/servlet/5.0/servlet-spec-5.0-SNAPSHOT.html
* https://jakarta.ee/specifications/authentication/2.0/authentication-spec-2.0-SNAPSHOT.html
* https://jakarta.ee/specifications/authorization/2.0/authorization-spec-2.0-SNAPSHOT.html

etc

**Comment by arjantijms** [comment]
> There is one mention to javax.xml.rpc.handler.MessageContext in the api for InvocationContext.
> It should be jakarta.xml.rpc.handler.MessageConte

Thanks! I've just updated it, and will now re-stage the API and update the javadocs in the PR.

**Comment by arjantijms** [comment]
Re-staged API: https://jakarta.oss.sonatype.org/content/repositories/staging/jakarta/interceptor/jakarta.interceptor-api/2.0.0/

**Comment by arjantijms** [comment]
Updated PR: https://github.com/jakartaee/specifications/pull/233/files#diff-839b8e3f2c49610e96e846712cb2c818R310

**Comment by hendrikebbers** [comment]
@arjantijms must this be updated somehow? Just trying to understand the workflows :)

**Comment by arjantijms** [comment]
@hendrikebbers yes indeed. This is easiest for myself todo. 

It's essentially running `mvn clean install` in the spec folder of interceptors, overriding the release to Final and the version to 2.0.0. Then checking out this repo, and copying it to the appropriate folder (see existing PR).

I'll do it now, though if you wanted to do it, the easiest way would be to checkout my PR branch and then do a PR against that.

**Comment by arjantijms** [comment]
Updated PR: https://github.com/jakartaee/specifications/pull/233/commits/d01e6bf88489dca8e76ef3defa9291b0400a45f1

**Comment by scottmarlow** [comment]
FYI, the RC1 Platform TCK link is https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-jakartaeetck-9.0.0-RC1.zip which has the same SHA `75b2493a117e7f8fe775cc7e2e8a605a203611895091f4bb9aaabed57f813392` as was used [to pass 100% of the Full Platform TCK yesterday](https://ci.eclipse.org/jakartaee-tck/job/jakartaee-tck/job/master/1165) (we are running again due to a system networking failure that occurred after the test run while collecting test results).  The [Jakarta EE 9 TCK results wiki page](https://github.com/eclipse-ee4j/jakartaee-tck/wiki/Jakarta-EE-9-TCK-results) will be updated after the run completes, not that you need to wait since I said, we already passed 100%.

[We also passed the Web Profile tests as well!](https://ci.eclipse.org/jakartaee-tck/job/eftl-jakartaeetck-certification-web/19/junit-reports-with-handlebars/testSuitesOverview.html).

This testing was done with https://download.eclipse.org/ee4j/glassfish/glassfish-6.0.0-RC2.zip.

**Comment by arjantijms** [comment]
Ballot thread: https://eclipse.org/lists/jakarta.ee-spec/msg01261.html

**Comment by kwsutter** [comment]
Per the discussion on our Spec Committee call on 11/11, I updated the reference to the EFTL TCK in the top level comment to be this:  https://download.eclipse.org/ee4j/jakartaee-tck/jakartaee9-eftl/promoted/jakarta-jakartaeetck-9.0.0.zip.

I based this update on the CCR and TCK Results which both referenced the "-RC1" version of the TCK.  This type of update was acceptable and expected by the Spec Committee.

**Comment by kazumura** [comment]
* On ballot completion, the specification committee mentor:
 - [x] adds this final checklist to the main PR.
 - [x] adds the `approved` label to the PRs, and sends out the Ballot Summary per this [template](#ballot-summary-email-template) to the [public Jakarta EE Specification Committee email list](jakarta.ee-spec@eclipse.org)
 - [ ] calculates the staged EFTL TCK signature and promotes it to the committee download area
  using the https://ci.eclipse.org/jakartaee-spec-committee/job/promote-release/ job. Manually editing the jenkins Build Information will help identify the build (ie. Mail 2.0 or CDI 3.0).
 - [ ] merges the specification (and apidocs) PRs, ensuring the "date:" field in the _index.md file has an appropriate value to allow publishing.
 - [ ] updates the specification page with the ballot results. This is normally done via a separate PR that should be reviewed, approved, and merged.
 - [ ] notifies the EMO of the ballot results by email to [emo@eclipse-foundation.org](emo@eclipse-foundation.org). Just forward the ballot summary note sent earlier to the public Spec Committee email list.
 - [x] creates an issue in the specification project that includes the following checklist for the specification project team:
 \<https://github.com/eclipse-ee4j/jsp-api/issues/195>


 ----
 - [ ] promotes api staging release promotes the specification api jars to maven central. An example release job script can be found here https://wiki.eclipse.org/MavenReleaseScript.
 - [ ] go through the merged jakarta.ee specification website page to verify all the links are valid.
 - [ ] approve the compatibility request.
 - [ ] The compatible implementation project/vendor MUST send an email to tck@eclipse.org for approval of the compatible implementation for trademark usage.
 - [ ] merge any final release branch as appropriate for the branch management for the project.

**Pull request merged**